### PR TITLE
Update Min supported version to iOS 11, tvOS 11 and Xcode 14.1 + Extension version Bump to 4.0.0

### PR DESCRIPTION
**Pull request by addb**
- Update Min supported version to iOS 11 and tvOS 11 (https://developer.apple.com/news/upcoming-requirements/?id=04252023a)
 - Bump major extension version.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: AEPEdge.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1326,7 +1331,7 @@
 				SUPPORTED_PLATFORMS = "appletvsimulator appletvos";
 				SWIFT_VERSION = 5.0;
 				TARGETED_DEVICE_FAMILY = 3;
-				TVOS_DEPLOYMENT_TARGET = 15.0;
+				TVOS_DEPLOYMENT_TARGET = 13.0;
```
Comment (kevinlind) [inline AEPEdge.xcodeproj/project.pbxproj:1334]: Can the tvOS test app use target 11.0?

File: AEPEdge.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1635,7 +1646,7 @@
 				SUPPORTED_PLATFORMS = "iphonesimulator iphoneos appletvsimulator appletvos";
 				SWIFT_VERSION = 5.0;
 				TARGETED_DEVICE_FAMILY = "1,2,3";
-				TVOS_DEPLOYMENT_TARGET = 10.0;
+				TVOS_DEPLOYMENT_TARGET = 11.0;
 				WATCHOS_DEPLOYMENT_TARGET = 4.0;
```
Comment (kevinlind) [inline AEPEdge.xcodeproj/project.pbxproj:1650]: Is is possible to remove `MACOSX_DEPLOYMENT_TARGET` and `WATCHOS_DEPLOYMENT_TARGET` from this functional test configuration?

File: Makefile
Comment (kevinlind) [inline Makefile:1]: In `test-ios`, update the `-destination` to `'platform=iOS Simulator,name=iPhone 14'`, which is what Core is now using.

File: Package.swift
Comment (kevinlind) [inline Package.swift:1]: Update Core dependency to "4.0.0"

File: Podfile
Comment (kevinlind) [inline Podfile:1]: Line 2, set `platform :ios, '11.0'`

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -51,8 +54,16 @@ jobs:
...
+    - name: Build Artifacts
+      if: ${{ github.event.inputs.action_tag == 'yes' }}
+      run: |
+        make archive
+        make zip
 
-    - uses: release-drafter/release-drafter@v5
+    - name: Create GH Release
+      id: create_release
+      uses: release-drafter/release-drafter@v5
```
Comment (kevinlind) [inline .github/workflows/release.yml:66]: Core is using this specific commit, `uses: release-drafter/release-drafter@fe52e97d262833ae07d05efaf1a239df3f1b5cd4`. Can you double check if this is required or if using `v5` is correct?

**Review by kevinlind — COMMENTED** [review]
You'll also need to update the CircleCI configuration `config.yml` to use xcode version `14.1.0` (from "13.4.0") plus update the simulators to specify version "16.1".
```
  prestart_ios_simulator:
    steps:
      - macos/preboot-simulator:
          platform: "iOS"
          version: "16.1"
          device: "iPhone 14"

  prestart_tvos_simulator:
    steps:
      - macos/preboot-simulator:
          platform: "tvOS"
          version: "16.1"
          device: "Apple TV"
```

File: AEPEdge.xcodeproj/project.pbxproj (hunk 1 above)
Comment (addb) [reply AEPEdge.xcodeproj/project.pbxproj:1334]: I have some APIs with min 13

File: .github/workflows/release.yml (hunk 3 above)
Comment (addb) [reply .github/workflows/release.yml:66]: Core reverted to v5.

File: README.md [hunk 4]
```diff
@@ -62,7 +62,7 @@ Alternatively, if your project has a `Package.swift` file, you can add AEPEdge d
 
 ```
```
Comment (cacheung) [inline README.md:63]: Need to change the Requirements to 
Xcode 14.1 (or newer)

**Comment by cacheung** [comment]
Need to update for test-SPM.sh version
https://github.com/addb/aepsdk-edge-ios/blob/xcode/Script/test-SPM.sh#L23

**Review by kevinlind — COMMENTED** [review]
Can you also update the versions in the SPM instructions in the getting-started guide?
https://github.com/adobe/aepsdk-edge-ios/blob/dev/Documentation/getting-started.md#download-the-edge-extension

**Review by kevinlind — APPROVED** [review]

**Comment by cacheung** [comment]
Under Readme file
Command line integration section, it mentions to use "make test".  There is no test action in make file.

File: AEPEdge.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -1499,13 +1506,15 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 10.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 11.0;
+				MARKETING_VERSION = "";
```
Comment (kevinlind) [inline AEPEdge.xcodeproj/project.pbxproj:1510]: Can you remove this empty MARKETING_VERSION? Looks like it was added by accident.

File: AEPEdge.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -1554,12 +1563,14 @@
...
```
Comment (kevinlind) [inline AEPEdge.xcodeproj/project.pbxproj:1567]: Remove this empty MARKETING_VERSION.

**Review by cacheung — APPROVED** [review]

**Review by kevinlind — APPROVED** [review]

**Pull request merged**